homeassistant/components/evohome.py

```diff
@@ -0,0 +1,667 @@
+"""Support for Honeywell evohome (EMEA/EU-based systems only).
+
+Support for a temperature control system (TCS, controller) with 0+ heating
+zones (e.g. TRVs, relays) and, optionally, a DHW controller.
+
+A minimal configuration.yaml is as as below:
+
+evohome:
+  username: !secret evohome_username
+  password: !secret evohome_password
+
+These config parameters are presented with their default values:
+
+# scan_interval: 180     # seconds, you can probably get away with 60
+# location_idx: 0        # if you have more than 1 location, use this
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/evohome/
+"""
+
+# Glossary
+# TCS - temperature control system (a.k.a. Controller, Master), which can
+# have up to 13 Slaves:
+#   0-12 Heating zones (a.k.a. Zone), and
+#   0-1 DHW controller, (a.k.a. Boiler)
+
+import logging
+from datetime import datetime, timedelta
+import requests
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ClimateDevice,
+
+    SUPPORT_OPERATION_MODE,
+    SUPPORT_AWAY_MODE,
+)
+
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+
+    EVENT_HOMEASSISTANT_START,
+    PRECISION_TENTHS,
+
+    STATE_OFF,
+    STATE_ON,
+    TEMP_CELSIUS,
+
+    HTTP_BAD_REQUEST,
+    HTTP_TOO_MANY_REQUESTS,
+    HTTP_SERVICE_UNAVAILABLE,
+)
+
+# These are HTTP codes commonly seen with this component
+#   HTTP_BAD_REQUEST = 400          # usually, bad user credentials
+#   HTTP_TOO_MANY_REQUESTS = 429    # usually, api limit exceeded
+#   HTTP_SERVICE_UNAVAILABLE = 503  # this is common with Honeywell's websites
+
+from homeassistant.core import callback
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['evohomeclient==0.2.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Usually, only the controller does client api I/O during update() to pull
+# current state - the exception is when zones pull their own schedules.
+# However, any entity can call methods that will (eventually) change state.
+PARALLEL_UPDATES = 0
+
+# these are specific to this component
+ATTR_UNTIL = 'until'
+
+DOMAIN = 'evohome'
+DATA_EVOHOME = 'data_' + DOMAIN
+DISPATCHER_EVOHOME = 'dispatcher_' + DOMAIN
+
+MIN_TEMP = 5
+MAX_TEMP = 35
+MIN_SCAN_INTERVAL = 180
+
+CONF_LOCATION_IDX = 'location_idx'
+
+# Validation of the user's configuration.
+CV_FLOAT = vol.All(vol.Coerce(float), vol.Range(min=MIN_TEMP, max=MAX_TEMP))
```

Comment:
What is this doing?

Reply:
Here, nothing - so I have taken it out.  However, it will be going back in later...

This PR was originally 2,500 lines. Balloob asked me to reduce it to the bare minimum for the initial PR, then add back some of the functionality in future PRs.  

FWIW, one example of using this: the user has the opportunity to tell the component if a non-default away temp is configured, because there is no way of extracting that data from the client api.

This is the reason why you might see oddities like the if-if that you rightly suggested could be a single if.

Reply:
:+1: Try to make sure any code that is not used for the items in _this_ PR is all removed. We don't want unused stuff sitting around

Reply:
Had another tidy-up.